### PR TITLE
stat: add example for LinearRegression

### PR DESCRIPTION
**Pull request by sbinet**
Please take a look.

### CODE REVIEW

File: stat/stat_example_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import (
+	"fmt"
+
+	"golang.org/x/exp/rand"
+
+	"gonum.org/v1/gonum/stat"
+)
+
+func ExampleLinearRegression() {
+
```
Comment (kortschak) [inline stat/stat_example_test.go]: Please delete this blank line, or fill it with some explanatory text.

File: stat/stat_example_test.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	fct := func(x float64) float64 {
+		return p0 + p1*x
+	}
+
+	for i := range xs {
+		xs[i] = float64(i)
+		ys[i] = fct(xs[i]) + 0.1*(rand.NormFloat64()-0.5)
+	}
+
+	origin := false
```
Comment (kortschak) [inline stat/stat_example_test.go]: It's worth noting what `origin` does here.

File: stat/stat_example_test.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	alpha, beta := stat.LinearRegression(xs, ys, nil, origin)
+	fmt.Printf("linreg(%v + %v*x)...\n", p0, p1)
+	fmt.Printf("alpha = %v\n", alpha)
+	fmt.Printf("beta =  %v\n", beta)
```
Comment (kortschak) [inline stat/stat_example_test.go]: Add in an R^2 assessment?

File: stat/stat_example_test.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	const (
+		n  = 100
+		p0 = 1
+		p1 = 3
+	)
+
+	xs := make([]float64, n)
+	ys := make([]float64, n)
+
+	fct := func(x float64) float64 {
```
Comment (btracey) [inline stat/stat_example_test.go]: Can we call this "line" and bring p0 and p1 in here?

File: stat/stat_example_test.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+		return p0 + p1*x
+	}
+
+	for i := range xs {
+		xs[i] = float64(i)
+		ys[i] = fct(xs[i]) + 0.1*(rand.NormFloat64()-0.5)
```
Comment (btracey) [inline stat/stat_example_test.go]: The `-0.5` just biases the random numbers to make the offset seem wrong. Remove it.

File: stat/stat_example_test.go [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+	}
+
+	origin := false
+	alpha, beta := stat.LinearRegression(xs, ys, nil, origin)
+	fmt.Printf("linreg(%v + %v*x)...\n", p0, p1)
+	fmt.Printf("alpha = %v\n", alpha)
```
Comment (btracey) [inline stat/stat_example_test.go]: "Estimated slope is:"

File: stat/stat_example_test.go (hunk 3 above)
Comment (btracey) [inline stat/stat_example_test.go]: "Estimated offset is:"?

**Comment by sbinet** [comment]
PTAL

**Comment by sbinet** [comment]
ping?

File: stat/stat_example_test.go [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+	line := func(x float64) float64 {
+		return 1 + 3*x
+	}
+
+	for i := range xs {
+		xs[i] = float64(i)
+		ys[i] = line(xs[i]) + 0.1*rand.NormFloat64()
+	}
+
+	// regression is not forced to pass through the origin
```
[the remark this answers is not in the document]
Comment (sbinet) [reply stat/stat_example_test.go]: done.

[the remark this answers is not in the document]
Comment (sbinet) [reply stat/stat_example_test.go]: oops. done.

**Pull request merged**